osu.Game/Rulesets/UI/GameplayCursorContainer.cs

```diff
@@ -17,11 +21,28 @@ public partial class GameplayCursorContainer : CursorContainer
         /// This holds the true visibility value.
         /// </summary>
         public Visibility LastFrameState;
+        public Vector2 TargetPosition = Vector2.Zero;
+        public Mod[] Mods;
+
+        public GameplayCursorContainer(Mod[] mods)
+        {
+            Mods = mods;
+        }
+
+        protected override bool OnMouseMove(MouseMoveEvent e)
+        {
+            TargetPosition = e.MousePosition;
+            return true;
+        }
 
         protected override void Update()
         {
             base.Update();
-            LastFrameState = State.Value;
+            Position = TargetPosition;
+            foreach (IModifiesCursorMovement modifier in Mods.OfType<IModifiesCursorMovement>())
+            {
+                Position = modifier.UpdatePosition(TargetPosition, (float)Time.Elapsed / 1000f);
+            }
```

Comment:
Something like this should be done in another class, see: [this](https://github.com/ppy/osu/blob/f050125c32bd1a623d3ce810785294954ae16894/osu.Game/Rulesets/UI/Playfield.cs#L380) or [this](https://github.com/ppy/osu/blob/b35e796d75153b53c1e450f407a48032aef1eddf/osu.Game/Rulesets/UI/DrawableRuleset.cs#L224)

Reply:
Am I being dumb, because it looks like the examples given don't have it done in another class.

Reply:
My point is just that I don't think a cursor should check if there are any mods and apply them if necessary. Imo this task shouldn't do a cursor.

Reply:
I'm not exactly sure what you meant, I moved it to Playfield. Is this correct?